### PR TITLE
Restore effect-preserving application

### PR DESCRIPTION
**Pull request by ewang12**
[Here](https://s3.amazonaws.com/stephan-misc/paper/branch-add-effect-preserving-app.pdf) is a link to the PDF generated from this PR.

### CODE REVIEW

File: main.tex [hunk 1]
```diff
@@ -107,6 +108,7 @@
           & $\evar$ & term variable \\
           & $\eabs{\tanno{\evar}{\tx}}{\eterm}$ & term abstraction \\
           & $\eapp{\eterm}{\eterm}$ & term application \\
+          & $\eappx{\eterm}{\eterm}$ & effect application \\
```
Comment (stepchowfun) [inline main.tex]: s/effect application/effect-preserving application/? or are we thinking of a new name for this

Comment (ewang12) [inline main.tex]: I just wanted it to match `T-EffectApplication` below, and `T-EffectPreservingApplication` seemed too long. But they don't have to match! Updating.

Comment (stepchowfun) [inline main.tex]: hm, but we already have a `T-EffectApplication` ~

File: main.tex [hunk 2]
```diff
@@ -283,6 +285,14 @@
       \end{prooftree}
 
       \begin{prooftree}
+          \AxiomC{$\tjudgment{\ccontext}{\dcontext}{\eterm_1}{\twithx{\ttype_1}{\xeffects_1}}$}
+          \AxiomC{$\tjudgment{\ccontext}{\dcontext}{\eterm_2}{\twithx{\parens{\tarrow{\twithx{\ttype_1}{\xeffects_4}}{\twithx{\ttype_2}{\xeffects_2}}}}{\xeffects_3}}$}
+          \AxiomC{$\xpre{\xeffects_1}{\xeffects_4}$}
+        \RightLabel{(\textsc{T-EffectApplication})}
+        \TrinaryInfC{$\tjudgment{\ccontext}{\dcontext}{\eappx{\eterm_2}{\eterm_1}}{\twithx{\ttype_2}{\xunion{\xeffects_2}{\xeffects_3}}}$}
+      \end{prooftree}
```
Comment (stepchowfun) [inline main.tex]: this rule LGTM. now I'm wondering why we need E_5 in the other application rule. why didn't we just put E1,E2,E3 in the conclusion?

Comment (stepchowfun) [inline main.tex]: (can we update that too)

Comment (ewang12) [inline main.tex]: I think we can just have E1,E2,E3 in the conclusion.

File: main.tex (hunk 1 above)
Comment (ewang12) [inline main.tex]: Oops, you're right.

**Pull request merged**